packages/exam-engine/src/dom-utils.ts

```diff
@@ -1,15 +1,81 @@
 export const NBSP = '\u00A0'
 
-export function parentElements(element: Element): Element[] {
+export function parentElements(element: Element, selector?: Selector): Element[] {
+  const predicate = selector != null ? mkPredicate(element, selector) : undefined
   const result = []
   while (element.parentElement != null) {
     element = element.parentElement
-    result.push(element)
+    if (predicate == null || predicate(element)) {
+      result.push(element)
+    }
   }
   return result
 }
 
-export function findChildElement(element: Element, predicate: (child: Element) => boolean): Element | undefined {
+type Selector = string | string[] | ((element: Element) => boolean)
```

Comment:
I'm thinking, whether some other name would be better for this type? I'm worried that developers will think of more generic CSS `Selector`s, which this API doesn't support.

Reply:
Query, perhaps